todo/views.py

```diff
@@ -1,49 +1,62 @@
-'''importing shortcuts'''
-from django.shortcuts import render, redirect
+'''importing shortcuts '''
+from django.shortcuts import render, get_object_or_404, redirect
 from django.contrib.auth.decorators import login_required
 from django.db.models import Q
-from.models import Todo
-# Creating views here.
+from .models import Todo
+from .forms import TodoForm
 
 def homepage(request):
     '''Returns the homepage.'''
     query = request.GET.get('q', '')
-    todos = Todo.objects.filter(Q(title__icontains=query) | Q(description__icontains=query))
+    todos = Todo.objects.filter(user=request.user).filter(Q(title__icontains=query) | Q(description__icontains=query))
```

Comment:
Search should only be applied when it has some value. Otherwise no need to apply that.